data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -108,6 +123,15 @@ protected long pos(T record) {
     return (Long) posAccessor.get(asStructLike(record));
   }
 
+  private ExecutorService readDeletesService(int workerPoolSize) {
+    return MoreExecutors.getExitingExecutorService(
+        (ThreadPoolExecutor) Executors.newFixedThreadPool(
+            workerPoolSize,
+            new ThreadFactoryBuilder()
+                .setNameFormat("Read-delete-Service-%d")
+                .build()));
+  }
```

Comment:
When is this executor service cleaned up?

Reply:
I think we chould create a custom executor and set `corePoolSize = 0`, `maximumPoolSize = poolSzie`, `workQueue = new LinkedBlockingQueue<Runnable>()` and let executor shutdown automatically. Otherwise, we need to call `shutdown` in `filter` or add `close` metod and call by `DeleterFilter` caller.

Reply:
I make this executor service as static. Now, different filter task can share the same executor service, and we don't need to consider when we should shutdown and cleaned up the executor service.